spec/controllers/pagarme_transactions_controller_spec.rb

```diff
@@ -0,0 +1,40 @@
+#encoding:utf-8
+require 'rails_helper'
+
+RSpec.describe PagarmeTransactionsController, type: :controller do
+  describe 'POST update status' do
+    context 'when is an invalid request' do
+      it 'returns a bad request' do
+        post :update_status
+
+        parsed_response = JSON.parse(response.body)
+
+        expect(response).to have_http_status :bad_request
+        expect(parsed_response).to eq('error' => 'invalid postback')
+      end
+    end
+
+    context 'when is a valid request' do
+      header = { 'HTTP_X_HUB_SIGNATURE' => 'signature' }
+      params = { id: 194330, current_status: 'refused' }
+      
+      before do
+        CatarseSettings[:email_payments] = 'financial@administrator.com'
+        create(:user, email: CatarseSettings[:email_payments])
+
+        allow(PagarMe::Postback).to receive(:valid_request_signature?)
+          .and_return(true)
+      end
+      
+      it 'returns success, change contribution state and create notification' do
+        contribution = create(:contribution, state: 'pending', payment_id: 194330)
+        
+        post :update_status, params, header
+
+        expect(response).to have_http_status :success
+        expect(contribution.reload.state).to eq('invalid_payment')
+        expect(contribution.notifications.size).to eq 1
```

Comment:
Não precisa testar cada caso, pois não é bem a responsabilidade desse teste. Em ordem de preferência eu faria isso:

1. Tentar se livrar do setup do email e se livrar dessa expectation. O teste ainda passa?
2. Achar uma transição que não dispare a notificação, assim o teste ficaria mais focado.
3. Deixar assim mesmo, mas pelo menos adicionar na descrição do "it" uma explicação bem curta.
4. Testar com mocks, assim não precisaria se preocupar com o setup do email. Mesmo assim essa é alternativa que prefiro menos (nesse caso específico), pois garante que os componentes funcionam bem em conjunto. Quando existe um teste de feature por cima, é :+1: usar mock (não é o caso aqui).